interface/modules/custom_modules/oe-module-weno/src/Services/TransmitProperties.php

```diff
@@ -30,23 +34,124 @@ class TransmitProperties
+            return;
+        }
+        // validated so create json object
+        $this->payload = $this->createJsonObject();
+    }
+
+    /**
+     * @param $value
+     * @return bool
+     */
+    public function isJson($value): bool
+    {
+        return is_string($value) && json_decode($value) !== null && (json_last_error() == JSON_ERROR_NONE);
+    }
+
+    /**
+     * @return false|string
+     */
+    public function getPayload(): false|string
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Generate a list of errors, warnings and info messages.
+     * All messages should be escaped and translated.
+     *
+     * @param $obj
+     * @return string[]
+     */
+    public function checkErrors($obj): array
+    {
+        // Initialize the error array
+        $error = ['errors' => '', 'warnings' => '', 'info' => '', 'string' => ''];
+        // Check if the input is a valid object
+        if (!is_object($obj)) {
+            return $error; // Return empty error array if the input is not an object
+        }
+        // Iterate through the object properties
+        foreach ($obj as $property => $value) {
+            // Skip 'errors' property and empty values
+            if ($property === 'errors' || empty($value)) {
+                continue;
+            }
+            // Extract error type and value
+            $type = '';
+            $v = '';
+            if (is_string($value)) {
+                $values = [$value];
+            } elseif ($this->isJson($value)) {
+                $values = json_decode($value, true);
+            } elseif (is_array($value)) {
+                $values = $value;
+            } else {
+                continue; // Skip non-array and non-string properties
+            }
+            // Iterate through the values
+            foreach ($values as $v) {
+                if (str_contains($v, "REQED")) {
+                    $type = 'errors';
+                } elseif (str_contains($v, "WARNS")) {
+                    $type = 'warnings';
+                } elseif (str_contains($v, "INFO")) {
+                    $type = 'info';
+                } else {
+                    continue; // Skip if no error type detected
+                }
+                // Add error to the respective error type if not already present
+                if (!str_contains($error[$type], $v)) {
+                    // Extract action from value
+                    $action = '';
+                    if (preg_match('/{([^}]*)}/', $v, $matches)) {
+                        $action = $matches[1];
+                        $v = str_replace('{' . $matches[1] . '}', '', $v);
+                    }
+                    // Append error with icon and onclick event
+                    $uid = attr_js($_SESSION['authUserID'] ?? 0);
+                    $action = attr_js($action);
+                    $error[$type] .= "<i onclick='renderDialog($action, $uid, event)' role='button' class='fas fa-info-circle mx-1'></i>$v<br>";
```

Comment:
we have an escaping master :+1: 

![kung-fu-baby-funny](https://github.com/openemr/openemr/assets/278968/6c4b00fe-c896-4c29-9c16-20b9935dc9bf)